File: backend/src/main/java/com/thirdhand/campusvault/entity/Payment.java
package com.thirdhand.campusvault.entity;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long paymentId;

    @ManyToOne
    @JoinColumn(name = "booking_id", nullable = false)
    private Booking booking;

    @Column(nullable = false)
    private BigDecimal amount;
    
    @Builder.Default
    private String status = "PENDING";

    @Column(unique = true)
    private String transactionId;

    private LocalDateTime paymentDate;

    private LocalDateTime createdAt;

    @PrePersist
    public void onCreate() {
        this.createdAt = LocalDateTime.now();
    }
}

File: frontend/web/types/payment.ts
export type PaymentStatus = "PENDING" | "SUCCESS" | "FAILED" | "REFUNDED";

export interface PaymentResponse {
  paymentId: number;
  bookingId: number;
  amount: number;
  status: PaymentStatus;
  paymentMethod: string;
  transactionRef: string;
  paidAt?: string;
  createdAt: string;
}
